field/src/types.rs

```diff
@@ -283,7 +283,7 @@ pub trait Field:
     }
 
     fn cyclic_subgroup_unknown_order(generator: Self) -> Vec<Self> {
-        let mut subgroup = Vec::new();
+        let mut subgroup = Vec::with_capacity(Self::TWO_ADICITY);
```

Comment:
Yes that's true, I had just assumed that in most cases we wouldn't create a group of just a few elements, hence the allocation (for the Goldilocks case this skips four reallocations for instance). But while doing it I also realized this method was actually never being used, is it a leftover? Happy to revert to the default case though.

Reply:
Yeah didn't end up using it, we could also just remove it. But if we keep it I think either the default or some explicit "larger default" like 256 would be good. Like if we added the Baby Bear field, it feels weird to start at 27 (since there's no subgroup of that order), or if we added Mersenne31, might be suboptimal to start at 1

Reply:
Given that we don't use it for now, I just reverted the changes on this one. We can always deal with this later on if its being used again.